broker.go

```diff
@@ -949,6 +972,91 @@ func (b *Broker) sendAndReceiveSASLOAuth(provider AccessTokenProvider) error {
 	return nil
 }
 
+func (b *Broker) sendAndReceiveSASLSCRAMv1(client SCRAMClient) error {
+	if err := b.sendAndReceiveSASLHandshake(b.conf.Net.SASL.Mechanism, SASLHandshakeV1); err != nil {
+		return err
+	}
+	c := b.conf.Net.SASL.SCRAMClient
```

Comment:
Is this a different client than argument? Also this is used quite a few time in this big function, more descriptive name will help readability.

Reply:
Renamed to `scramClient`.